src/Plugin.php

```diff
@@ -86,6 +93,45 @@ public static function upload_mime_types(array $mimes) {
   }
 
   /**
+   * Removes SVG image size attributes from the content.
+   *
+   * @implements the_content
+   */
+  public static function the_content($content) {
+    $content = preg_replace_callback('@<img [^>]*src=[^>]+?.svg[^>]*>@', function ($matches) {
+      return static::removeSizeAttributes($matches[0]);
+    }, $content);
+    return $content;
+  }
+
+  /**
+   * Removes size attributes from SVG images.
+   *
+   * @implements wp_get_attachment_metadata
+   * @implements image_send_to_editor
+   * @implements post_thumbnail_html
+   */
+  public static function removeSvgSizeAttributes($data, $post_id, $attachment_id = NULL) {
+    if ($data && get_post_mime_type($attachment_id ?? $post_id) === 'image/svg+xml') {
+      $data = static::removeSizeAttributes($data);
+    }
+    return $data;
+  }
+
+  /**
+   * Removes size attributes from a given string or array.
+   */
+  public static function removeSizeAttributes($data) {
+    if (is_array($data)) {
+      unset($data['width'], $data['height']);
```

Comment:
I agree for bit more complicated readability. But isn't it better because it avoids repetition? If we change it to two separate functions we have to distinguish upfront which function to call which I wanted to avoid extensive nesting. Optionally we could do an early return. Also I'd like to stick with a helper function as the regex is used 2 times.

```php
public static function removeSvgSizeAttributes($data, $post_id, $attachment_id = NULL) {
  if ($data && get_post_mime_type($attachment_id ?? $post_id) === 'image/svg+xml') {
    if (is_array($data)) {
      unset($data['width'], $data['height']);
    }
    else {
      $data = static::removeSizeAttributes($data);
    }
  }
  return $data;
}
```
or
```php
public static function removeSvgSizeAttributes($data, $post_id, $attachment_id = NULL) {
  if (!$data || get_post_mime_type($attachment_id ?? $post_id) !== 'image/svg+xml') {
    return $data;
  }
  if (is_array($data)) {
    unset($data['width'], $data['height']);
  }
  else {
    $data = static::removeSizeAttributes($data);
  }
  return $data;
}
```

I don't like that both solutions much which I'd rather stick with the overladed function but add a comment. What do you think?